x/smart/stream_dialer.go

```diff
@@ -253,18 +255,50 @@ func (f *StrategyFinder) testDialer(ctx context.Context, dialer transport.Stream
 		testCtx, cancel := context.WithTimeout(ctx, f.TestTimeout)
 		defer cancel()
 
+		// Dial
+
 		testConn, err := dialer.DialStream(testCtx, testAddr)
 		if err != nil {
 			f.logCtx(ctx, "🏁 failed to dial: '%v' (domain: %v), duration=%v, dial_error=%v ❌\n", transportCfg, testDomain, time.Since(startTime), err)
 			return err
 		}
+
+		// TLS Connection
+
 		tlsConn := tls.Client(testConn, &tls.Config{ServerName: testDomain})
+		defer tlsConn.Close()
 		err = tlsConn.HandshakeContext(testCtx)
-		tlsConn.Close()
 		if err != nil {
 			f.logCtx(ctx, "🏁 failed TLS handshake: '%v' (domain: %v), duration=%v, handshake=%v ❌\n", transportCfg, testDomain, time.Since(startTime), err)
 			return err
 		}
+
+		// HTTPS Get
+
+		req, err := http.NewRequestWithContext(testCtx, "GET", "https://"+testDomain, nil)
+		if err != nil {
+			return fmt.Errorf("failed to create HTTP request: %w", err)
+		}
+
+		if err := req.Write(tlsConn); err != nil {
+			f.logCtx(ctx, "🏁 failed to write HTTP request: '%v' (domain: %v), duration=%v, error=%v ❌\n", transportCfg, testDomain, time.Since(startTime), err)
+			return err
+		}
+
+		resp, err := http.ReadResponse(bufio.NewReader(tlsConn), req)
+		if err != nil {
+			f.logCtx(ctx, "🏁 failed to read HTTP response: '%v' (domain: %v), duration=%v, error=%v ❌\n", transportCfg, testDomain, time.Since(startTime), err)
+			return err
+		}
+		defer resp.Body.Close()
+
+		// Validate that the response status is not empty, but don't require the content to be anything specific.
+		// We cannot validate the response code, since many bare domains return i.e. 301 redirects, or even 404
+		if resp.Status == "" {
```

Comment:
I don't think we need to check the status if we got a valid response above.

Why would the status be empty but not fail above?